src/context/auth.tsx

```diff
@@ -0,0 +1,94 @@
+import { useMutation } from '@tanstack/react-query'
+import { destroyCookies, setCoookies, DefaultError } from '@/utils'
+import {
+  createContext,
+  ReactNode,
+  useState,
+  useContext,
+  useEffect,
+} from 'react'
+import { parseCookies } from 'nookies'
+import { SignInInput } from '@/types/user'
+import { Either, isRight } from 'fp-ts/Either'
+import { signInMutation, type SignInResponseOutput } from './signInMutation'
+
+type Status = 'loggedOut' | 'loggedIn' | 'idle'
+
+type ContextProps = {
+  status: Status
+  signIn: (values: SignInInput) => void
+  isLoading: boolean
+  signOut: () => void
+  data: Either<DefaultError, SignInResponseOutput> | undefined
```

Comment:
Porque usar `undefined` e não option do fp-ts?

Reply:
I fix in this [commit](https://github.com/joaooliveiradev/conduit-frontend/pull/35/commits/dfb7f6ff11ff59605c6f9b54956484203da980a4)